### PR TITLE
initial

### PR DESCRIPTION
**Pull request by MaksymKos**
https://maksymkos.github.io/react_list-of-goods-hooks/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +17,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum TypeOfSort {
+  name = 'name',
+  length = 'length',
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:22]:
```suggestion
  NAME = 'name',
  LENGTH = 'length',
  NONE = 'none',
```
general convention make fields of enums in upper case
and better to set for none sort

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +17,104 @@ export const goodsFromServer = [
...
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState('');
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:55]:
```suggestion
  const [sortType, setSortType] = useState(TypeOfSort.NONE);
```

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +17,104 @@ export const goodsFromServer = [
...
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button is-info',
+            { 'is-light': sortType !== TypeOfSort.name },
+          )}
+          onClick={() => setSortType(TypeOfSort.name)}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:73]: consider to make one hanler factory for all sort fields which will take sortType and return arrow function.
```
const handleSort = (sortType: TypeOfSort) => () => setSortType(sortType);
```
```suggestion
          onClick={handleSort(TypeOfSort.NAME)}
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +17,104 @@ export const goodsFromServer = [
...
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button is-success',
+            { 'is-light': sortType !== TypeOfSort.length },
+          )}
+          onClick={() => setSortType(TypeOfSort.length)}
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:84]:
```suggestion
          onClick={handleSort(TypeOfSort.LENGTH)}
```

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]

**Review by etojeDenys — APPROVED** [review]
lgtm

**Pull request closed without merging**